### PR TITLE
error on missing locale

### LINKED ISSUE #989: Show a warning for "default_locale" with no correspondent in _locales folder
Steps to reproduce:
1. Submit a web-extension containing a `default_value` set in the manifest.json that has no correspondent in **_locales** folder

Expected results:
A warning is displayed at validation.

Actual results:
No validation errors. User can make it to step 3 of submission process, where **is blocked** because it has no name field and cannot go forward because that field is mandatory.

Notes/Issues:
Verified on FF49(Win 7). Issue is reproducing all around AMO servers.
Screenshots for this issue: 
ok
submit
required

### PR DESCRIPTION
**Pull request by andymckay**
* fixes #989

### CODE REVIEW

File: src/messages/manifestjson.js [hunk 1]
```diff
@@ -107,3 +107,13 @@ export function manifestPropMissing(property) {
 
 export const PROP_NAME_MISSING = manifestPropMissing('name');
 export const PROP_VERSION_MISSING = manifestPropMissing('version');
+
+export const NO_MESSAGES_FILE = {
+  code: 'NO_MESSAGES_FILE',
+  legacyCode: null,
+  message: _('The "default_locale" is missing localisations.'),
```
Comment (tofumatt) [inline src/messages/manifestjson.js]: I think the Mozilla style guide says to use American English, at least as the default locale.

File: src/linter.js [hunk 2]
```diff
@@ -235,7 +235,7 @@ export default class Linter {
               var manifestParser = new ManifestJSONParser(
                 json,
                 this.collector,
-                {selfHosted: this.config.selfHosted},
+                {selfHosted: this.config.selfHosted, files:this.io},
```
Comment (tofumatt) [inline src/linter.js]: `files` seems like the wrong variable name here; I'd expect an array of files, but this is an `IO` object.

There really isn't a great variable name I can think of, so I'd say rename this to at least `io` so it's easier to trace through.

File: src/linter.js [hunk 3]
```diff
@@ -250,7 +250,7 @@ export default class Linter {
       .then((addonMetadata) => {
         this.addonMetadata = addonMetadata;
 
-        // The type must be explcitly defined. This behaviour differs the
+        // The type must be explicitly defined. This behaviour differs the
```
Comment (tofumatt) [inline src/linter.js]: Thanks! 😄

File: src/linter.js (hunk 2 above)
Comment (tofumatt) [inline src/linter.js]: Oh and very minor nitpick but we've been doing `files: this.io` (note the space after colon).

We should have stylelint rules for it but we seem to lack them. I've filed #1069 for that.

File: src/messages/manifestjson.js (hunk 1 above)
Comment (andymckay) [inline src/messages/manifestjson.js]: Grr 😄

File: src/linter.js (hunk 2 above)
Comment (andymckay) [inline src/linter.js]: Fair enough.

**Pull request merged**